src/base/bittorrent/session.cpp

```diff
@@ -1093,6 +1176,61 @@ void Session::configure(libtorrent::settings_pack &settingsPack)
     settingsPack.set_bool(libt::settings_pack::enable_lsd, isLSDEnabled());
 }
 
+void Session::configurePeerClasses()
+{
+    libt::ip_filter f;
+    f.add_rule(libt::address_v4::from_string("0.0.0.0")
+               , libt::address_v4::from_string("255.255.255.255")
+               , 1 << libt::session::global_peer_class_id);
+#if TORRENT_USE_IPV6
+    f.add_rule(libt::address_v6::from_string("::0")
+               , libt::address_v6::from_string("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff")
+               , 1 << libt::session::global_peer_class_id);
+#endif
+    if (ignoreLimitsOnLAN()) {
+        f.add_rule(libt::address_v4::from_string("10.0.0.0")
+                   , libt::address_v4::from_string("10.255.255.255")
+                   , 1 << libt::session::local_peer_class_id);
+        f.add_rule(libt::address_v4::from_string("172.16.0.0")
+                   , libt::address_v4::from_string("172.16.255.255")
+                   , 1 << libt::session::local_peer_class_id);
+        f.add_rule(libt::address_v4::from_string("192.168.0.0")
+                   , libt::address_v4::from_string("192.168.255.255")
+                   , 1 << libt::session::local_peer_class_id);
+        f.add_rule(libt::address_v4::from_string("169.254.0.0")
+                   , libt::address_v4::from_string("169.254.255.255")
+                   , 1 << libt::session::local_peer_class_id);
+        f.add_rule(libt::address_v4::from_string("127.0.0.0")
+                   , libt::address_v4::from_string("127.255.255.255")
+                   , 1 << libt::session::local_peer_class_id);
+#if TORRENT_USE_IPV6
+        f.add_rule(libt::address_v6::from_string("fe80::")
+                   , libt::address_v6::from_string("febf:ffff:ffff:ffff:ffff:ffff:ffff:ffff")
```

Comment:
hmmm, this is weird, can't wrap my head to understand how this was generated.
IPv6 link-local: `fe80::/10`
IPv6 Unique Local Addresses: `fc00::/7`